crates/wysiwyg/src/tests/test_lists.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(test)]
+
+use crate::tests::testutils_composer_model::{cm, tx};
+
+use crate::{ComposerModel, ToHtml};
+
+#[test]
+fn test_ordered_list() {
+    let mut model = cm("|");
+    model.create_ordered_list();
+    assert_eq!(tx(&model), "<ol><li>|</li></ol>");
+    replace_text(&mut model, "abcd");
+    assert_eq!(tx(&model), "<ol><li>abcd|</li></ol>");
+    model.enter();
+    assert_eq!(tx(&model), "<ol><li>abcd</li><li>\u{200b}|</li></ol>");
+    replace_text(&mut model, "efgh");
+    assert_eq!(tx(&model), "<ol><li>abcd</li><li>\u{200b}efgh|</li></ol>");
+}
+
+#[test]
+fn test_unordered_list() {
```

Comment:
I suspect we need tests for where some other text already exists.

I think the cursor will be in the wrong place in that case.

I am happy for this code to be merged, and then we fix that case. There are going to be LOADS of corner cases!